pkg/controller/cincinnati/cincinnati_controller.go

```diff
@@ -214,39 +234,117 @@ func handleCACertStatus(reqLogger logr.Logger, status *cv1beta1.CincinnatiStatus
 	if err != nil && errors.IsNotFound(err) {
 		m := fmt.Sprintf("image.config.openshift.io not found for (Name: %v, Namespace: %v)", defaults.ImageConfigName, "")
-		handleCACertStatus(reqLogger, &instance.Status, "EnsureAdditionalTrustedCAFailed", m)
-		return nil
+		handleCACertStatus(reqLogger, &instance.Status, "FindAdditionalTrustedCAFailed", m)
+		return nil, nil
 	} else if err != nil {
-		return err
+		return nil, err
 	}
 
 	if image.Spec.AdditionalTrustedCA.Name == "" {
 		m := fmt.Sprintf("image.config.openshift.io.Spec.AdditionalTrustedCA.Name not found for image (Name: %v, Namespace: %v)", defaults.ImageConfigName, "")
-		handleCACertStatus(reqLogger, &instance.Status, "EnsureAdditionalTrustedCAFailed", m)
-		return nil
+		handleCACertStatus(reqLogger, &instance.Status, "FindAdditionalTrustedCAFailed", m)
+		return nil, nil
 	}
 
 	// Search for the ConfigMap in openshift-config
 	sourceCM := &corev1.ConfigMap{}
 	err = r.client.Get(ctx, types.NamespacedName{Name: image.Spec.AdditionalTrustedCA.Name, Namespace: openshiftConfigNamespace}, sourceCM)
 	if err != nil && errors.IsNotFound(err) {
 		m := fmt.Sprintf("Found image.config.openshift.io.Spec.AdditionalTrustedCA.Name but did not find expected ConfigMap (Name: %v, Namespace: %v)", image.Spec.AdditionalTrustedCA.Name, openshiftConfigNamespace)
-		handleCACertStatus(reqLogger, &instance.Status, "EnsureAdditionalTrustedCAFailed", m)
-		return err
+		handleCACertStatus(reqLogger, &instance.Status, "FindAdditionalTrustedCAFailed", m)
+		return nil, err
 	} else if err != nil {
-		return err
+		return nil, err
 	}
 
 	if _, ok := sourceCM.Data[NameCertConfigMapKey]; !ok {
 		m := fmt.Sprintf("Found ConfigMap referenced by ImageConfig.Spec.AdditionalTrustedCA.Name but did not find key 'cincinnati-registry' for registry CA cert in ConfigMap (Name: %v, Namespace: %v)", image.Spec.AdditionalTrustedCA.Name, openshiftConfigNamespace)
 		handleCACertStatus(reqLogger, &instance.Status, "EnsureAdditionalTrustedCAFailed", m)
+		return nil, nil
+	}
+
+	return sourceCM, nil
+}
+
+func (r *ReconcileCincinnati) ensurePullSecret(ctx context.Context, reqLogger logr.Logger, instance *cv1beta1.Cincinnati, resources *kubeResources) error {
+	sourcePS, err := r.findPullSecret(ctx, reqLogger, instance, resources)
+	if err != nil {
+		return err
+	} else if sourcePS == nil {
+		return nil
+	}
+
+	// Set Cincinnati instance as the owner and controller
+	if err := controllerutil.SetControllerReference(instance, resources.pullSecret, r.scheme); err != nil {
+		return err
+	}
+
+	if err := r.ensureSecret(ctx, reqLogger, resources.pullSecret); err != nil {
+		handleErr(reqLogger, &instance.Status, "EnsureSecretFailed", err)
+		return err
+	}
+
+	return nil
+}
+
+func (r *ReconcileCincinnati) ensureAdditionalTrustedCA(ctx context.Context, reqLogger logr.Logger, instance *cv1beta1.Cincinnati, resources *kubeResources) error {
+	sourceCM, err := r.findTrustedCAConfig(ctx, reqLogger, instance, resources)
```

Comment:
Same as above, why call this here?

Reply:
Similar to above except `findTrustedCAConfig` handles different scenarios.

Potential errors from `findTrustedCAConfig` are:
_Found image.config.openshift.io.Spec.AdditionalTrustedCA.Name but did not find expected ConfigMap (Name: %v, Namespace: %v)_
_Failed to GET the object_ (API timeout or error)

The question we're trying to answer here is:
_Were we able to find a non empty configmap named `image.config.openshift.io.Spec.AdditionalTrustedCA.Name` in openshift-config?_
**No there was an error** -> error
**Yes** and **Configmap == nil** -> return nil
**Yes** and **Configmap != nil** -> continue.  Ensure configmap.